src/sys/fallocate.rs

```diff
@@ -0,0 +1,20 @@
+use libc;
+use libc::{off_t, c_int};
+use std::os::unix::io::RawFd;
+use {Errno, Result};
+
+libc_bitflags!(
+    pub struct FallocateFlags: libc::c_int {
+        FALLOC_FL_KEEP_SIZE;
+        FALLOC_FL_PUNCH_HOLE;
+        FALLOC_FL_COLLAPSE_RANGE;
+        FALLOC_FL_ZERO_RANGE;
+        FALLOC_FL_INSERT_RANGE;
+        FALLOC_FL_UNSHARE_RANGE;
+    }
+);
+
+pub fn fallocate(fd: RawFd, mode: c_int, offset: off_t, len: off_t) -> Result<c_int> {
```

Comment:
Also needs docs, see the other PRs or the source for how they're commonly written.